docs/bookmarks.md

```diff
@@ -0,0 +1,67 @@
+
+# Debugging
+
+The UI shows a bookmark on the document when any user has bookmarked the document. When a user clicks the bookmark CLUE toggles that user's bookmark on that document. If the user doesn't have a bookmark yet, then one is added. This user bookmark will be added even if the document is already showing a bookmark from a different user. All of this can be confusing.
+
+To expose this confusing logic the debug option `bookmarks` can be used. It will show a code above the bookmark on the thumbnail of the document. Like `U1 T1 O1`.
+- `U1` means the current user has a bookmark for the document
+- `T1` means 1 teacher other than you has a bookmark for the document
+- `O1` means 1 other user in the class has a bookmark for the document
+The numbers will increase if more than one teacher or other user has bookmarked the document.
+
+Additionally the `bookmarks` debug option will console log a message when the bookmark is click. This message includes all of the bookmarks on the document prior to the click.
+
+# How to represent multiple bookmarks on a single doc
+
+If we want to continue to show bookmarks made by anyone to everyone here is my proposal:
+- have 4 groups of bookmarks in the sort work tab: you, teacher, others(or other students), none
+- add some indication to the bookmark to indicate who has bookmarked it: you, teacher, others. You can see a rough demo of this with the debugging option described above.
+
+## Why you might want to bookmark something the Teacher already has:
+You want to bookmark some document even though the teacher did, because the teacher tends to removing bookmarks after that document isn't relevant anymore, and you want to keep track of it. So you still need to toggle your own bookmark of this document.
+
+## Why the teacher might not want to see bookmarks from other class members
+They might curate the bookmarks for the class so if anyone can bookmark stuff the teacher will lose control. Without the separate grouping, the teacher would have no way to hide other student bookmarks.
+
+## Why have a teacher bookmark section
+If there is more than one teacher in the class, each of them might want to control this list of bookmarks. Even with improved display of the bookmarks, this case isn't handled well. If Teacher A has bookmarked a document, then Teacher B cannot un-bookmark it. This means the document would still continue to show in the teacher bookmark section.
+
+# Improvements
+- rename all references to `stars` to `bookmarks`
+- remove the flag of `starred` on the star in Firebase, and just use the presence of the star.
+- move the bookmarks to Firestore, they fit the firestore model a better because we can query them for the ones owned by a user and ones owned by a teacher.
```

Comment:
```suggestion
- move the bookmarks to Firestore, they fit the firestore model better because we can query them for the ones owned by a user and ones owned by a teacher.
```